OneCommunityGlobal/HGNRest
# Jatin create a function to send email to user's who completed hours but missed summary

# Description
<img width="582" alt="Screenshot 2024-09-08 at 1 42 24 PM" src="https://github.com/user-attachments/assets/bee7c2e9-e6ff-4499-a0cc-8d90753cc8af">

## Related PRS (if any):
To test this backend PR you need to checkout the dev branch in  frontend .

## Main changes explained:
- Created a function completeHoursAndMissedSummary in src/helpers/userHelper.js
- Added a CronJob in src/cronjobs/userProfileJobs.js

##Before Testing
To send an email using the Gmail API with OAuth2 and nodemailer, follow the video below and set it up.
https://www.youtube.com/watch?v=-rcRf7yswfM&t=10s&ab_channel=MafiaCodes
Note: Publish your OAuthConsentScreen After creating it. This video is only for the purpose of learning how to get the credentials.
Fill in the these variable values from the above setup in the .env file of the backend.
sendEmail=true
REACT_APP_EMAIL =
REACT_APP_EMAIL_CLIENT_ID =
REACT_APP_EMAIL_CLIENT_SECRET =
REACT_APP_EMAIL_REFRESH_TOKEN =
REACT_APP_EMAIL_CLIENT_REDIRECT_URI = https://developers.google.com/oauthplayground

## How to test:
1. check into current branch
2. do `npm install` and `npm run build` to run this PR locally
3. Go to  src/server.js. Import completeHoursandMissedSummary  and call the function as described in the image below.
4. Go to src/helpers/userHelper.js. Find completeHoursandMissedSummary.
5. Replace the recipient email where the emailSender function is called with your email which you have access to (To avoid sending emails to other users).
6. Enter a console statement in the if statement enclosing the emailSender function, so one can know when a user is found and email has been sent.
7.  Note: The code may take some time to send an email as it checks all user's who have completed their hours but missed their summary.

## Frontend Repository
1. Log in as owner/admin any user.
2. Add hours which are equivalent to total hours as intangible time. Select them as tangible.
3. Make sure the summary is not submitted.

## Screenshots or videos of changes:
<img width="389" alt="Screenshot 2024-09-14 at 5 25 48 PM" src="https://github.com/user-attachments/assets/d6601b4e-f49b-46a4-a0db-83b7ab3e34f8">

https://github.com/user-attachments/assets/1d080ada-7bf5-46c9-b7ef-c6d6f303661c
